instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/Serializer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.protocols.core.OperationInfo;
+import software.amazon.awssdk.protocols.core.ProtocolMarshaller;
+import software.amazon.awssdk.protocols.json.AwsJsonProtocolFactory;
+import software.amazon.awssdk.utils.IoUtils;
+
+class Serializer {
+
+  private final ProtocolMarshaller<SdkHttpFullRequest> protocolMarshaller;
+
+  Serializer() {
+    protocolMarshaller =
+        AwsJsonProtocolFactory.builder()
+            .clientConfiguration(
+                SdkClientConfiguration.builder()
+                    .option(SdkClientOption.ENDPOINT, URI.create("http://empty"))
+                    .build())
+            .build()
+            .createProtocolMarshaller(
+                OperationInfo.builder()
+                    .hasPayloadMembers(true)
+                    .httpMethod(SdkHttpMethod.POST)
+                    .build());
+  }
+
+  @Nullable
+  private String serialize(SdkPojo sdkPojo) {
+    Optional<ContentStreamProvider> optional =
+        protocolMarshaller.marshall(sdkPojo).contentStreamProvider();
+    return optional
+        .map(
+            csp -> {
+              try {
+                return IoUtils.toUtf8String(csp.newStream());
+              } catch (IOException e) {
+                return null;
+              }
+            })
+        .orElse(null);
+  }
+
+  private String serialize(Collection<Object> collection) {
+    return collection.stream().map(this::serialize).collect(Collectors.joining(","));
+  }
+
+  @Nullable
+  String serialize(Object target) {
```

Comment:
Confirming target can't be null?

Reply:
Would require a member of a collection (field type) to be NULL. IMO not possible, but will add a guard, just to stay safe.

Reply:
Let's add `@Nullable` - since we're interfacing with another SDK we need to expect the worse and consider it correct.